src/Parameter.h

```diff
@@ -96,6 +96,15 @@ class Parameter {
         *((T *)(scalar_address())) = val;
     }
 
+    /** If the parameter is a scalar parameter, set its current
+     * value. Only relevant when jitting */
+    NO_INLINE void set_scalar(const Type &val_type, halide_scalar_value_t val) {
```

Comment:
pass by const-ref might be better here

Reply:
Why? Pretty sure halide_scalar_value_t is always passed in a register on recent architecture.